tests/integration/submission/test_versioning.py

```diff
@@ -0,0 +1,280 @@
+            client,
+            headers,
+            data_fnames2=test_fnames + [sur_filename])
+
+    assert_message = 'Unable to create nodes: {}'.format(
+        [entity for entity in resp.json['entities'] if entity['errors']]
+    )
+    assert resp.status_code in (200, 201), assert_message
+
+    with open(os.path.join(DATA_DIR, sur_filename), 'r') as f:
+        sur_json = json.loads(f.read())
+
+    sur_uuid = [
+        entity['id']
+        for entity in resp.json['entities']
+        if entity['type'] == 'submitted_unaligned_reads'
+    ][0]
+
+    file_metadata = {
+        'did': sur_uuid,
+        'file_size': sur_json['file_size'],
+        'file_name': sur_json['file_name'],
+        'md5sum': sur_json['md5sum'],
+    }
+
+    return file_metadata
+
+
+def test_create_data_file_entity(
+        client, indexd_server, indexd_client, pg_driver, cgci_blgsp, submitter):
+    """
+    Create a new node in the database.
+
+    Success conditions:
+        - Node created in database
+        - Entry created in indexd
+        - Entry in indexd has pertinent metadata
+        - Entry in indexd has no version
+            - Exactly one entry in indexd can have no version number
+            - Nodes that have not been released do not have a version
+            associated with them
+    """
+
+    # node created in database
+    data_file = data_file_creation(
+        client,
+        submitter,
+        method='post',
+        sur_filename='submitted_unaligned_reads.json',
+    )
+
+    # entry created in indexd
+    indexd_doc = indexd_client.get(data_file['did'])
+    assert indexd_doc
+
+    # entry in indexd has metadata
+    assert indexd_doc.did == data_file['did']
+    assert indexd_doc.size == data_file['file_size']
+    assert indexd_doc.file_name == data_file['file_name']
+    assert indexd_doc.hashes['md5'] == data_file['md5sum']
+
+    # entry in indexd has no version
+    assert indexd_doc.version is None
+
+
+def test_update_data_file_entity(
+        app, client, indexd_server, indexd_client, pg_driver, cgci_blgsp, submitter):
+    """
+    Update an existing node in the database.
+
+    The API allows a user to update a node with new information. This new
+    information can be a partial update or a full node replace. On an update
+    the indexd document will get deleted and recreated with the same old
+    fields and the new fields supplied by the user.
+
+    Success conditions:
+        - Node already in database
+        - Entry already in indexd
+        - Update to node in indexd does not have a version associated
+        - Indexd entry has new information supplied by the user
+        - Exactly one entry in indexd can have no version number
+        - Base IDs between the two same nodes should be the same
+        - Cannot update a node if it is in state submitted
+    """
+
+    # simulate a project api that lets you do this
+    app.config['CREATE_REPLACEABLE'] = True
```

Comment:
Just want to plug something fancy that I did :)
https://github.com/uc-cdis/sheepdog/blob/release/versioning/tests/integration/conftest.py#L138
used like this: https://github.com/uc-cdis/sheepdog/blob/release/versioning/tests/integration/submission/test_upload.py#L649